types/ramda/index.d.ts

```diff
@@ -2069,6 +2077,9 @@ export function pipe<TArgs extends any[], R1, R2>(
     f1: (...args: TArgs) => R1,
     f2: (a: R1) => R2,
 ): (...args: TArgs) => R2;
+// For inference, so should be first. Works with generics and overloads
+// Useful for seeing which overload is picked by default
+export function pipe<F extends (...args: any[]) => unknown>(f1: F): F;
```

Comment:
I don't think we should add this overload, this will swallow the above overloads' ts errors.

Reply:
Maybe we can rewrite `pipe`'s type in one more elegant way。

Reply:
Maybe. Are you sure they will swallow the above overloads' errors? This one only takes one argument, right? So the only error it should swallow is the one for the overload below.

If you mean that it swallows it because it's below all the others (not sure about this one), I could try just moving it to the top of the overload list